src/test/java/com/ceritoxi/util/MathTest.java

```diff
@@ -0,0 +1,157 @@
+		for (int i = 0; i < expectedResult2.length; i++) {
+			assertEquals(expectedResult2[i], result2[i]);
+		}
+		for (int i = 0; i < expectedResult3.length; i++) {
+			assertEquals(expectedResult3[i], result3[i]);
+		}
+		for (int i = 0; i < expectedResult4.length; i++) {
+			assertEquals(expectedResult4[i], result4[i]);
+		}
+    }
+	
+	public void testMathIsPrime() {
+		assertEquals(true, Math.isPrime(2));
+		assertEquals(false, Math.isPrime(1));
+		assertEquals(false, Math.isPrime(0));
+		assertEquals(false, Math.isPrime(-300));
+		assertEquals(false, Math.isPrime(-2));
+		assertEquals(true, Math.isPrime(6857));
+		assertEquals(true, Math.isPrime(2L));
+		assertEquals(false, Math.isPrime(1L));
+		assertEquals(false, Math.isPrime(0L));
+		assertEquals(false, Math.isPrime(-300L));
+		assertEquals(false, Math.isPrime(-2L));
+		assertEquals(true, Math.isPrime(6857L));
+    }
+	
+	public void testMathNextPrime() {
+		assertEquals(2, Math.nextPrime(0));
+		assertEquals(2, Math.nextPrime(1));
+		assertEquals(3, Math.nextPrime(2));
+		assertEquals(5, Math.nextPrime(3));
+		assertEquals(5, Math.nextPrime(4));
+		assertEquals(2L, Math.nextPrime(0L));
+		assertEquals(2L, Math.nextPrime(1L));
+		assertEquals(3L, Math.nextPrime(2L));
+		assertEquals(5L, Math.nextPrime(3L));
+		assertEquals(5L, Math.nextPrime(4L));
+    }
+	
+	public void testMathPrimeIndex() {
+		boolean expectedResult1 = false;
+		boolean expectedResult2 = false;
+		try {
+			Math.primeIndex(1);
+		} catch (IllegalArgumentException e) {
+			expectedResult1 = true;
+		}
+		try {
+			Math.primeIndex(0);
+		} catch (IllegalArgumentException e) {
+			expectedResult2 = true;
+		}
+		assertTrue(expectedResult1);
+		assertTrue(expectedResult2);
+		assertEquals(0, Math.primeIndex(2));
+		assertEquals(1, Math.primeIndex(3));
+		assertEquals(2, Math.primeIndex(5));
+		assertEquals(3, Math.primeIndex(7));
+	}
+	
+	public void testMathLargestPrimeFactorOf() {
+		assertEquals(6857, Math.largestPrimeFactorOf(6857));
+		assertEquals(5, Math.largestPrimeFactorOf(100000));
+		assertEquals(0, Math.largestPrimeFactorOf(0));
+		assertEquals(0, Math.largestPrimeFactorOf(-100000));
+    }
+
+    public void testMathIsPalindrome() {
+		assertEquals(true, Math.isPalindrome(0));
+		assertEquals(true, Math.isPalindrome(9889));
+		assertEquals(true, Math.isPalindrome(-9889));
+		assertEquals(false, Math.isPalindrome(10));
+		assertEquals(false, Math.isPalindrome(-224));
+		assertEquals(false, Math.isPalindrome(6857));
+    }
+	
+	public void testMathLargestPalindromeMadeFromTwoXDigitNumberThatAreLessThenSix() {
+		assertEquals(906609, Math.largestPalindromeMadeFromTwoXDigitNumberThatAreLessThenSix(3));
+    }
+	
+	public void testMathPositiveIntegerPower() {
+		assertEquals(1, Math.positiveIntegerPower(10, 0));
+		assertEquals(10, Math.positiveIntegerPower(10, 1));
+		assertEquals(25, Math.positiveIntegerPower(5, 2));
+		assertEquals(10000, Math.positiveIntegerPower(-10, 4));
+    }
+	
```

Comment:
Höhöhö, you did not test the most fucked up functions you cheeky bastard :). 

Gimme the test for greatestCommonDivisor and largestProductInAStringOfDigitsMadeFormXAdjacentDigits and all the rest. 
Also the Util class is not tested at all

Reply:
Sure it will happen :DD